### ISSUE TITLE
Setting center on MGLMapView initialization is at wrong coordinate

### ISSUE THREAD

**Issue by rmnblm** [mapbox/mapbox-gl-native#6753]
**Platform:** iOS 10.1
**Mapbox SDK version:** 3.4.0-beta.1

### Steps to trigger behavior

``` swift 
private func addMapView() {
    mapView = MGLMapView()
    mapView.delegate = self
    mapView.styleURL = URL(string: "mapbox://styles/mapbox/outdoors-v9")!
    mapView.setCenter(startupCoordinate, zoomLevel: startupZoomLevel, animated: false)
    view.addSubview(mapView)

    let tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(mapViewTap(recognizer:)))
    mapView.addGestureRecognizer(tapGestureRecognizer)
}
```

### Expected behavior

Center coordinate is where I expect it, here:

![unadjustednonraw_thumb_2af0](https://cloud.githubusercontent.com/assets/5942764/19519692/5cce4ef0-960d-11e6-9623-c3e50b665bde.jpg)

### Actual behavior

Center coordinate is at wrong position, here:

![unadjustednonraw_thumb_2aef](https://cloud.githubusercontent.com/assets/5942764/19519689/5a105730-960d-11e6-886f-745c1f2909c9.jpg)

### Solution

Setting the center coordinate in the `mapViewDidFinishLoadingMap`-delegate function resolves the issue but it's not what I expect.

``` swift
func mapViewDidFinishLoadingMap(_ mapView: MGLMapView) {
    mapView.setCenter(startupCoordinate, zoomLevel: startupZoomLevel, animated: false)
}
```

**Comment by friedbunny** [comment]
I can’t reproduce this — several of our examples set the center coordinate in `-viewDidLoad` without issue ([like this one](https://www.mapbox.com/ios-sdk/examples/custom-style/)).

Can you confirm that coordinates you’re setting are correct?

**Comment by rmnblm** [comment]
I used exactly the same coordinates in both examples above. One points me somewhere in the North Sea and one points me to the correct location.

I tried the same today and it seems now that it's working correctly which is more strange than if it would behave like yesterday. I will close this issue for now and reopen in if it occurs again.

Maybe something I didn't explicitly tell: I am using SnapKit to set some constraints which I reordered. Don't know if this caused the weird behaviour I described.
